From: Steve Dickson <SteveD@redhat.com>
Date: Fri, 12 Feb 2010 15:40:46 -0500
Subject: [nfs] Fix a bug in nfs_fscache_release_page()
Message-id: <4B75767E.5050201@RedHat.com>
Patchwork-id: 23249
O-Subject: [RHEL6] [PATCH 1/2] NFS: Fix a bug in nfs_fscache_release_page()
Bugzilla: 563938

commit 2c1740098c708b465e87637b237feb2fd98f129a
Author: Trond Myklebust <Trond.Myklebust@netapp.com>
Date:   Mon Feb 8 09:32:27 2010 -0500

    NFS: Fix a bug in nfs_fscache_release_page()

    Not having an fscache cookie is perfectly valid if the user didn't mount
    with the fscache option.

    This patch fixes http://bugzilla.kernel.org/show_bug.cgi?id=15234

    Signed-off-by: Trond Myklebust <Trond.Myklebust@netapp.com>
    Acked-by: David Howells <dhowells@redhat.com>
    Cc: stable@kernel.org

Signed-off-by: Aristeu Rozanski <arozansk@redhat.com>

diff --git a/fs/nfs/fscache.c b/fs/nfs/fscache.c
index fa58800..237874f 100644
--- a/fs/nfs/fscache.c
+++ b/fs/nfs/fscache.c
@@ -354,12 +354,11 @@ void nfs_fscache_reset_inode_cookie(struct inode *inode)
  */
 int nfs_fscache_release_page(struct page *page, gfp_t gfp)
 {
-	struct nfs_inode *nfsi = NFS_I(page->mapping->host);
-	struct fscache_cookie *cookie = nfsi->fscache;
-
-	BUG_ON(!cookie);
-
 	if (PageFsCache(page)) {
+		struct nfs_inode *nfsi = NFS_I(page->mapping->host);
+		struct fscache_cookie *cookie = nfsi->fscache;
+
+		BUG_ON(!cookie);
 		dfprintk(FSCACHE, "NFS: fscache releasepage (0x%p/0x%p/0x%p)\n",
 			 cookie, page, nfsi);
 
